owncloudApp/src/main/java/com/owncloud/android/presentation/settings/more/SettingsMoreFragment.kt

```diff
@@ -108,11 +110,15 @@ class SettingsMoreFragment : PreferenceFragmentCompat() {
         // Feedback
         if (moreViewModel.isFeedbackEnabled()) {
             prefFeedback?.setOnPreferenceClickListener {
-                val feedbackMail = getString(R.string.mail_feedback)
-                val feedback = "Android v" + BuildConfig.VERSION_NAME + " - " + getString(R.string.prefs_feedback)
-
-                requireActivity().sendEmail(email = feedbackMail, subject = feedback)
-                true
+                if (moreViewModel.getFeedbackMail().isNotEmpty()) {
+                    val feedbackMail = getString(R.string.mail_feedback)
+                    val feedback = "Android v" + BuildConfig.VERSION_NAME + " - " + getString(R.string.prefs_feedback)
+                    requireActivity().sendEmail(email = feedbackMail, subject = feedback)
+                    true
+                } else {
+                    requireActivity().openFeedbackDialog()
+                    true
+                }
```

Comment:
This is repeated code, the same as the one that appears in `DrawerActivity`. Maybe you can add it to `ActivityExt` directly?

Reply:
In the first instance I decided to remove the `getFeedbackMail()` from the `DrawerViewModel` class, and take the value of the resource directly from `ActivityExt`. The problem is that when I did the test I couldn't find a way to mock this value, so I adapted the code a bit so that the test passed correctly. In conclusion I simply passed this function as a parameter to `feedbackBehaviour()`.